packages/algjulia-interop/src/decapodes-service/diagram.jl

```diff
@@ -0,0 +1,149 @@
+## DIAGRAM BUILDING
+
+function add_to_pode! end
+export add_to_pode!
+
+function add_to_pode!(d::SummationDecapode, 
+        vars::Dict{String, Int}, # mapping between UUID and ACSet ID
+        model::Any, 
+        content::AbstractDict,
+        nc::Dict{Int, String},
+        ::ObTag)
+
+    # indexes the model by UUID
+    model_element = model.data[content[:over][:content]] 
+    
+    # checks if the cell is an anonymous (intermediate) variable.
+    # if so, we increment the intermediate variable counter and make an intermediate variable name. 
+    # otherwise we use the existing name of the given content.
+    name = if isempty(content[:name])
+        id = isempty(keys(nc)) ? 1 : maximum(keys(nc)) + 1
+        push!(nc, id => "")
+        Symbol("•$id")
+    else
+        Symbol(content[:name])
+    end 
+    id = add_part!(d, :Var, name=name, type=nameof(model_element))
+    push!(vars, content[:id] => id)
+    return d
+end
+
+function Base.nameof(model::Model, content::AbstractDict)
+    if isnothing(content[:over])
+        :no_name
+    else
+        Symbol(model.data[content[:over][:content]].name)
+    end
+end
+
+# TODO we are restricted to Op1
+function add_to_pode!(d::SummationDecapode,
```

Comment:
This and the other method `add_to_pode!` have a bunch of arguments, but most of them are already in the `DecapodeDiagram` struct defined below. I would suggest making these be methods `add_to_pode!(d::DecapodeDiagram, ...)`. This will require slightly refactoring your `Diagram` code below so that the `DecapodeDiagram` is created *before* you have to call `add_to_pode!` but I think it will ultimately simplify this code and make it easier to read.